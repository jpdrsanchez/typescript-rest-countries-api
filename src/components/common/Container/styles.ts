import styled from 'styled-components'

export const Wrapper = styled.div`
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  position: relative;

  @media (max-width: 35.9375em) {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  @media (min-width: 36em) {
    max-width: 34rem;
  }

  @media (min-width: 48em) {
    max-width: 45rem;
  }

  @media (min-width: 62em) {
    max-width: 60rem;
  }

  @media (min-width: 75em) {
    max-width: 72rem;
  }

  @media (min-width: 82em) {
    max-width: 80rem;
  }
`
