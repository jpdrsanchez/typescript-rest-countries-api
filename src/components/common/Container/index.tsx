import { ReactNode } from 'react'

import * as S from './styles'

type ContainerProps = {
  className?: string
  children: ReactNode
}

const Container = ({ className, children }: ContainerProps) => {
  return <S.Wrapper className={className}>{children}</S.Wrapper>
}

export default Container
