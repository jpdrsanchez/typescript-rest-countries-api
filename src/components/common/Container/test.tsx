import { render, screen } from '@testing-library/react'
import Container from '.'

describe('<Container />', () => {
  it('Renders children correctly', () => {
    render(
      <Container>
        <div data-testid="children" />
        <div data-testid="children" />
        <div data-testid="children" />
      </Container>
    )

    const children = screen.getAllByTestId('children')

    expect(children).toHaveLength(3)
  })

  it('Renders with correct class names', () => {
    const { container } = render(
      <Container className="testClass">
        <div />
      </Container>
    )

    expect(container.firstChild).toHaveClass('testClass')
  })

  it('Renders correctly', () => {
    const { container } = render(
      <Container>
        <div />
      </Container>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
