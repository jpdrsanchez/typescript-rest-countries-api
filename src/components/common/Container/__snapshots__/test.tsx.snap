// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Renders correctly 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  position: relative;
}

@media (max-width:35.9375em) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media (min-width:36em) {
  .c0 {
    max-width: 34rem;
  }
}

@media (min-width:48em) {
  .c0 {
    max-width: 45rem;
  }
}

@media (min-width:62em) {
  .c0 {
    max-width: 60rem;
  }
}

@media (min-width:75em) {
  .c0 {
    max-width: 72rem;
  }
}

@media (min-width:82em) {
  .c0 {
    max-width: 80rem;
  }
}

<div
  class="c0"
>
  <div />
</div>
`
